File: sbt-plugin/src/main/scala/play/soap/sbtplugin/PlayGenerator.scala
/*
 * Copyright (C) 2015-2016 Lightbend Inc. <https://www.lightbend.com>
 */
package play.soap.sbtplugin

/**
 * Adds common attributes for all source generation templates
 */
trait PlayGenerator {

  /**
   * Set the Play attributes
   */
  def setPlayAttributes() = {
    setAttribute("version", Version.pluginVersion)
    setAttribute("fullversion", Version.name + " " + Version.pluginVersion)
    setAttribute("name", Version.name)
    setAttribute("generatorclass", SbtWsdl.getClass.getName)
  }

  /**
   * Set an attribute
   */
  def setAttribute(name: String, value: AnyRef)
}

File: plugin/src/main/java/play/soap/plugin/PlayGenerator.java
/*
 * Copyright (C) from 2022 The Play Framework Contributors <https://github.com/playframework>, 2011-2021 Lightbend Inc. <https://www.lightbend.com>
 */
package play.soap.plugin;

/** Adds common attributes for all source generation templates */
public interface PlayGenerator {

  /** The name of the plugin */
  String NAME = "Play SOAP";

  default String version() {
    return PlayGenerator.class.getPackage().getImplementationVersion();
  }

  /** Set the Play attributes */
  default void setPlayAttributes() {
    setAttribute("version", version());
    setAttribute("fullversion", NAME + " " + version());
    setAttribute("name", NAME);
    setAttribute("generatorclass", "PlayGenerator");
  }

  /** Set an attribute */
  void setAttribute(String name, Object value);
}
